### PR TITLE
XP Channel for wildfly-jar-maven-plugin

### PR DESCRIPTION
**Pull request by tommaso-borgato**
This MR adds the `bootablejar.profile.channels` profile which add the option to run bootable jar test using a channel for assembling the botable jar applocation

Please make sure your PR meets the following requirements:
- [x] Pull Request contains a description of the changes
- [x] Pull Request does not include fixes for multiple issues/topics
- [x] Code is formatted, imports ordered, code compiles and tests are passing
- [x] Link to the passing job is provided
- [x] Code is self-descriptive and/or documented
- [x] Description of the tests scenarios is included (see #46)

### CODE REVIEW

**Review by fabiobrz — COMMENTED** [review]
I notice you added a configuration for the `maven-surefire-plugin`  to each module. Why is that? The TS was configured already for executing such plugin, so what is the reason for adding more?

That being said, I don't see the need for an additional profile. I think we can simply modify the `wildfly-jar-maven-plugin` execution in the `ts.botable`profile, for it to use channels and that's all. WDYT?

**Comment by tommaso-borgato** [comment]
> I notice you added a configuration for the `maven-surefire-plugin` to each module. Why is that? The TS was configured already for executing such plugin, so what is the reason for adding more?
> 
> That being said, I don't see the need for an additional profile. I think we can simply modify the `wildfly-jar-maven-plugin` execution in the `ts.botable`profile, for it to use channels and that's all. WDYT?

@fabiobrz  I went for distinct profiles in CI, e.g. 
```
if [[ "x${CHANNEL_MANIFEST_GAV}" != "x" ]]; then
  echo "[INFO] Using Channel Manifest GAV: ${CHANNEL_MANIFEST_GAV}"
  generate_channel_manifest_gav_mvn_params
  ACTIVATION_PROPERTY_NAME=ts.bootable.channels
else
  ACTIVATION_PROPERTY_NAME=ts.bootable
fi
```
that's the reason for the duplication.... You had a smart idea, If you prefer, I can change it and activate two profiles: one for bootable jar + one for using a channel fo rbootable jar

**Comment by fabiobrz** [comment]
> I went for distinct profiles in CI, e.g.
> 
> ```
> if [[ "x${CHANNEL_MANIFEST_GAV}" != "x" ]]; then
>   echo "[INFO] Using Channel Manifest GAV: ${CHANNEL_MANIFEST_GAV}"
>   generate_channel_manifest_gav_mvn_params
>   ACTIVATION_PROPERTY_NAME=ts.bootable.channels
> else
>   ACTIVATION_PROPERTY_NAME=ts.bootable
> fi
> ```
> 
> that's the reason for the duplication. 

Ok thanks. 
Your snippet is valid from the automation PoV. It would be good to add a few lines to the documentation, where we mention the same difference from a user's PoV, e.g.: 

_you might want to use `ts.bootable` with WildFly since it doesn't have a channel manifest, while using `ts.bootable.channels` allows to configure a channel manifest, which is the supported EAP (and EAP XP) way_ 

or something like that. But anyway, the `ts.bootable` option isn't documented at all yet, so IMO just logging an issue would be fine ATM.

> If you prefer, I can change it and activate two profiles: one for bootable jar + one for using a channel fo rbootable jar

I think it is now clear what you've done here and why. Let's just file an issue documenting the Bootable JAR related profiles.

**Review by fabiobrz — COMMENTED** [review]
Hi @tommaso-borgato - Just reviewed the latest changes now, and dropped a cuple of comments.

File: microprofile-config/pom.xml [hunk 1]
```diff
@@ -95,6 +95,89 @@
                         </executions>
                     </plugin>
 
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
```
[the remark this answers is not in the document]
Comment (tommaso-borgato) [reply microprofile-config/pom.xml]: I leveraged your initial suggestion and added an incremental profile for using channel with bootable jar

File: pom.xml [hunk 2]
```diff
@@ -80,6 +80,13 @@
         <!-- Bootable JAR -->
         <version.org.wildfly.jar.plugin>9.0.0.Final</version.org.wildfly.jar.plugin>
         <version.org.bitbucket.b_c.jose4j>0.7.4</version.org.bitbucket.b_c.jose4j>
+
+        <!-- XP Channel manifest GAV -->
+        <channel-manifest.groupId>org.jboss.eap.channels</channel-manifest.groupId>
+        <channel-manifest.artifactId>eap-xp-5.0</channel-manifest.artifactId>
+        <!-- Leave this property empty since it activates the use of the XP channel in bootable jar builds -->
+        <channel-manifest.version></channel-manifest.version>
```
[the remark this answers is not in the document]
Comment (tommaso-borgato) [reply pom.xml:88]: I leveraged your initial suggestion and added an incremental profile for using channel with bootable jar

[the remark this answers is not in the document]
Comment (fabiobrz) [reply pom.xml:88]: Nice, thanks!

**Review by fabiobrz — APPROVED** [review]
Thanks for addressing my remarks, @tommaso-borgato 
Now we just need the name of the Jenkins job and the run number in here, to provide evidence that tests are passing. 
Would you mind to add them to the description?

**Review by honza-kasik — APPROVED** [review]

**Comment by tommaso-borgato** [comment]
CI Job run eap-8.x-microprofile-testing-pipeline-tborgato#9

**Comment by tommaso-borgato** [comment]
@fabiobrz , @honza-kasik  can we merge this one?

**Comment by fabiobrz** [comment]
> @fabiobrz , @honza-kasik can we merge this one?

Thanks @tommaso-borgato - changes LGTM, and the CI run shows just unrelated failures. Merging.

**Pull request merged**